Assets/QA/Tests/Core Platform Menu/AddScenesToBuild.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+[InitializeOnLoad]
+public class AddScenesToBuild : EditorWindow
+{
+    static AddScenesToBuild()
+    {
+        EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+    }
+
+    [MenuItem("Tools/Add All Scenes to Build")]
+    private static void AddAllScenesToBuildExcludingXboxAndXR()
+    {
+        // Get all available scenes in the project
+        string[] sceneGuids = AssetDatabase.FindAssets("t:Scene");
+        string[] scenePaths = new string[sceneGuids.Length];
+
+        for (int i = 0; i < sceneGuids.Length; i++)
+        {
+            scenePaths[i] = AssetDatabase.GUIDToAssetPath(sceneGuids[i]);
+        }
+
+        // Filter out scenes in folders containing "xbox" or "xr"
+        List<string> filteredScenePaths = new List<string>();
+        foreach (string path in scenePaths)
+        {
+            if (!IsPathInExcludedFolder(path))
+            {
+                filteredScenePaths.Add(path);
+            }
+        }
+
+        // Update the build settings
+        EditorBuildSettingsScene[] buildScenes = new EditorBuildSettingsScene[filteredScenePaths.Count];
+
+        for (int i = 0; i < filteredScenePaths.Count; i++)
+        {
+            buildScenes[i] = new EditorBuildSettingsScene(filteredScenePaths[i], true);
+        }
+
+        EditorBuildSettings.scenes = buildScenes;
+
+        Debug.Log("All scenes (excluding Xbox and XR) added to build settings.");
+    }
+
+    private static bool IsPathInExcludedFolder(string path)
+    {
+        // Specify folder names to exclude
+        string[] excludedFolders = { "xbox", "xr" };
+
+        // Check if the path or any part of it contains any of the excluded folder names
+        foreach (string folder in excludedFolders)
+        {
+            if (path.ToLower().Contains(folder.ToLower()))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static void OnPlayModeStateChanged(PlayModeStateChange state)
+    {
+        if (state == PlayModeStateChange.ExitingPlayMode)
+        {
+            // Save the build settings before entering play mode
```

Comment:
Isn't this the wrong way round. Sounds like its exiting play mode and we wouldn't want to be saving during that phase.

Reply:
Leaving play mode in Editor would always clear my added build scenes so the next time I enter play mode I couldn't use the scene loader